rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala

```diff
@@ -206,3 +205,40 @@ object Uniform {
   def apply(from: Real, to: Real): Continuous =
     standard.scale(to - from).translate(from)
 }
+
+/**
+  * A mixture distribution composed of continuous distributions
+  *
+  * @param pmf A Map with keys representing component beta distributions and values corresponding to the probabilities of those components.
+  * @param support The support of the new mixture distribution.
+  */
+final case class ContinuousMixture(pmf: Map[Continuous, Real], support: Support)
+    extends Continuous {
+  def realLogDensity(t: Real): Real =
+    Real
+      .sum(pmf.toList.map {
+        case (dist, prob) =>
+          (dist.realLogDensity(t) + prob.log).exp
+      })
+      .log
+
+  def generator: Generator[Double] =
+    Categorical(pmf).generator.flatMap { d =>
+      d.generator
+    }
+}
+
+object ContinuousMixture {
+
+  /**
+    * Allows the creation of a mixture distribution without specifying the support, using the support of one of its components.
+    * @param supportDistribution A distribution (and its weight) in the mixture with support which is the union of all supports in the mixture.
+    * @param otherDistributions The other distributions, in a Map(distribution -> distr weight) form.
+    */
+  def apply(supportDistribution: (Continuous, Real),
```

Comment:
should we compute the support distribution's weight with 1.0 - the sum of the other dists weight?

Reply:
I'm not sure if that'd be less or more confusing... we could always have two `apply` methods of course.